src/compiler/types.rs

```diff
@@ -112,6 +134,26 @@ impl From<Expression> for Node {
     }
 }
 
+#[derive(Debug)]
+pub struct BinaryExpression {
+    pub _node: BaseNode,
+    pub left: Box<Expression>,
```

Comment:
Add `Box` indirection to avoid circularity

Reply:
Is this a situation where it'd have an unbounded size without this?

Reply:
Ya since `Expression` has an enum variant that contains a `BinaryExpression` I think trying to have `BinaryExpression` contain an `Expression` would make the size unbounded